src/template_part.py

```diff
@@ -0,0 +1,81 @@
+import json
+import os
+
+
+class TemplatePart:
+    """
+    Class to manage a template part for ElabFTW.
+    """
+
+    def __init__(self, template_part_file):
+        """
+        Initializes the TemplatePart object.
+
+        Parameters:
+            template_part_file (str): Path to the JSON file.
+        """
+        self.template_part_file = template_part_file
+        self.template_part = self.read_template_part(template_part_file)
+
+    @staticmethod
+    def read_template_part(template_parts_file):
+        """
+        Reads a JSON template part file and validates its structure.
+
+        Parameters:
+            template_parts_file (str): Path to the JSON file.
+
+        Returns:
+            dict: Content of the JSON file.
+
+        Raises:
+            ValueError: If the structure is invalid.
+        """
+        try:
+            with open(template_parts_file, 'r') as f:
+                template_parts_content = json.load(f)
+                TemplatePart.check_structure(template_parts_content)
+                return template_parts_content
+        except KeyError as e:
+            raise ValueError(f"Invalid template file: missing key {e}")
+
+    @staticmethod
+    def check_structure(template_parts_file_content):
+        """
+        Validates the structure of a template part.
```

Comment:
this is very specific to elab... it should appear as such! (name of the variables? of the method?)

also, same as above, are you checking the validity of a TemplatePart or of a Template? both should be implemented...